pkgs/flojoy/flojoy/flojoy_cloud.py

```diff
@@ -1,475 +1,344 @@
+import datetime
 import json
-from typing import Generic, Optional, TypeVar
+import os
+from typing import Literal, Optional, TypeVar, overload
 
 import numpy as np
-import pandas as pd
 import requests
-from PIL import Image as PILImage
-from pydantic import BaseModel, validator
-
-from flojoy import DataContainer
-from flojoy.data_container import (
-    DataFrame,
-    Grayscale,
-    Image,
-    Matrix,
-    OrderedPair,
-    OrderedTriple,
-    Scalar,
-    Vector,
-)
-from flojoy.utils import PlotlyJSONEncoder
+from pydantic import BaseModel, ConfigDict, TypeAdapter
+from pydantic.alias_generators import to_camel
 
 
 class NumpyEncoder(json.JSONEncoder):
     """json encoder for numpy types."""
 
-    def default(self, obj):
-        if isinstance(obj, np.integer):
-            return int(obj)
-        elif isinstance(obj, np.floating):
-            return float(obj)
-        elif isinstance(obj, np.ndarray):
-            return obj.tolist()
-        return json.JSONEncoder.default(self, obj)
+    def default(self, o):
+        if isinstance(o, np.integer):
+            return int(o)
+        elif isinstance(o, np.floating):
+            return float(o)
+        elif isinstance(o, np.ndarray):
+            return o.tolist()
+        return json.JSONEncoder.default(self, o)
 
 
-DataC = TypeVar("DataC")
+MeasurementType = Literal["boolean", "dataframe"]
 
 
-class DefaultModel(BaseModel, Generic[DataC]):
-    ref: str
-    dataContainer: dict
-    workspaceId: str
-    location: str
-    note: str
-
-
-class OrderedPairModel(DefaultModel[DataC], Generic[DataC]):
-    data: Optional[DataC]
-
-    @validator("dataContainer")
-    def type_must_match(cls, dc):
-        assert dc["type"] == "OrderedPair", "dataContainer type does not match."
-        return dc
+class Boolean(BaseModel):
+    type: Literal["boolean"] = "boolean"
+    passed: bool
 
-    @validator("dataContainer")
-    def keys_must_match(cls, dc):
-        assert "x" in dc, 'dataContainer does not contain "x" dataset.'
-        assert "y" in dc, 'dataContainer does not contain "y" dataset.'
-        assert isinstance(dc["x"], list), '"x" dataset is not a list'
-        assert isinstance(dc["y"], list), '"y" dataset is not a list'
-        return dc
-
-
-class OrderedTripleModel(DefaultModel[DataC], Generic[DataC]):
-    data: Optional[DataC]
 
-    @validator("dataContainer")
-    def type_must_match(cls, dc):
-        assert dc["type"] == "OrderedTriple", "dataContainer type does not match."
-        return dc
+class Dataframe(BaseModel):
+    type: Literal["dataframe"] = "dataframe"
+    dataframe: dict
+
+
+MeasurementData = Boolean | Dataframe
+
+
+T = TypeVar("T", bound=BaseModel)
+U = TypeVar("U")
+
+
+class CloudModel(BaseModel):
+    model_config = ConfigDict(alias_generator=to_camel)
```

Comment:
Cloud server uses camelCase for fields because it's in javascript, python client uses snake_case because python, this just does the name conversions automatically